tool/gravity/cli/config.go

```diff
@@ -1352,6 +1357,33 @@ func checkLocalAddr(addr string) error {
 		addr, strings.Join(availableAddrs, ", "))
 }
 
+// checkOverlayNet verifies that the host network address is not in the range of the serviceCIDR or podCIDR
+func checkOverlayNet(advertiseAddr string, serviceCIDR string, podCIDR string) error {
+	_, serviceSubnet, err := net.ParseCIDR(serviceCIDR)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	ip := net.ParseIP(advertiseAddr)
+	if ip == nil {
+		return trace.BadParameter("unable to parse advertiseAddr(%s) ", advertiseAddr)
+	}
+	if serviceSubnet.Contains(ip) {
+		return trace.BadParameter("advertiseAddr(%s) is in range of the serviceCIDR %s",
```

Comment:
Keep in mind that these errors will be shown in a terminal so try to make them easily understandable by an average user and helpful by providing possible resolution advice, for example:

```
The selected advertise address %v conflicts with the service network CIDR range %v.
Please specify a different service CIDR via --service-cidr flag.
```

Same for other error messages here.